### PR TITLE
README.md: improve subtitle explanations

### PR DESCRIPTION
**Pull request by mikkovedru**
### Improve subtitle explanations

I had to test it myself in order to understand how `--sub-langs` works. For example, I found out that the subtitle "en-GB" will NOT be downloaded by either `--sub-langs "en*"` or by  `--sub-langs "en-*"`.

It would be better to decrease the confusion by improving the ReadMe.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
📝🛠️🎨

Update `README.md` to clarify and improve the documentation for the `--sub-format` and `--sub-langs` options. Use consistent syntax and formatting for the subtitle options.

> _`--sub-format` docs_
> _Updated with clearer syntax_
> _Autumn of confusion_

### Walkthrough
*  Update and reformat the documentation for the subtitle options in `README.md` ([link](https://github.com/yt-dlp/yt-dlp/pull/8248/files?diff=unified&w=0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L896-R906))

</details>

### CODE REVIEW

**Comment by gamer191** [comment]
Love this PR, as someone who once wrote an inaccurate bug report about en-GB not working with en*

I will have a think about if it can be worded better though, since it still doesn't explicitly mention that wildcards should be proceeded by a dot. Any ideas?

Note: this PR shouldn't be merged in it's current form, since it doesn't modify the options.py file. That should be trivial to fix though (I'm not on my computer right now, but I'll fix it next time I come across this PR, if no one else has)

**Comment by mikkovedru** [comment]
> I will have a think about if it can be worded better though, since it still doesn't explicitly mention that wildcards should be proceeded by a dot. Any ideas?

English is not my native language so I am sure it can be worded even better. Feel free to suggest (or make them directly).

When I encountered this gotcha, I was trying to find a better place to put it into. Some kind of "section where subtitle options are being explained" would be a natural choice (with expanded explanations), but no such section exists. Therefore, instead of creating a new one, I decided to append the help text and make it as small as I could, even if that made me a little dirty inside. :-)

Great catch about the `options.py`, thank you! It has now been fixed.

**Comment by mikkovedru** [comment]
@gamer191 
Could you please review and approve the push?

File: yt_dlp/options.py [hunk 1]
```diff
@@ -876,13 +876,14 @@ def _alias_callback(option, opt_str, value, parser, opts, nargs):
     subtitles.add_option(
         '--sub-format',
         action='store', dest='subtitlesformat', metavar='FORMAT', default='best',
-        help='Subtitle format; accepts formats preference, e.g. "srt" or "ass/srt/best"')
+        help='Subtitle format; accepts formats preference, separated by "/" e.g. "srt" or "ass/srt/best"')
```
Comment (bashonly) [inline yt_dlp/options.py:879]:
```suggestion
        help='Subtitle format; accepts formats preference separated by "/", e.g. "srt" or "ass/srt/best"')
```

File: yt_dlp/options.py [hunk 2]
```diff
@@ -876,13 +876,14 @@ def _alias_callback(option, opt_str, value, parser, opts, nargs):
...
     subtitles.add_option(
         '--sub-langs', '--srt-langs',
         action='callback', dest='subtitleslangs', metavar='LANGS', type='str',
         default=[], callback=_list_from_options_callback,
         help=(
             'Languages of the subtitles to download (can be regex) or "all" separated by commas, e.g. --sub-langs "en.*,ja". '
+            'Example: "en.*" will download both subtitles "en" and "en-GB", while "en" will only download the former. '
```
Comment (bashonly) [inline yt_dlp/options.py:886]:
```suggestion
            'Languages of the subtitles to download (can be regex) or "all" separated by commas, e.g. --sub-langs "en.*,ja" '
            '(where "en.*" is a regex pattern that matches "en" followed by 0 or more of any character). '
```

**Review by bashonly — APPROVED** [review]

**Pull request closed without merging**